http/src/main/java/io/micronaut/http/uri/UriMatchTemplate.java

```diff
@@ -202,6 +198,29 @@ protected UriTemplateParser createParser(String templateString, Object... parser
         return new UriMatchTemplateParser(templateString, this);
     }
 
+    private boolean isRoot() {
+        CharSequence rawSegment = null;
+        for (PathSegment segment : segments) {
+            if (segment.isVariable()) {
+                if (!segment.isQuerySegment()) {
+                    return false;
+                }
+            } else {
+                if (rawSegment == null) {
+                    rawSegment = segment;
+                } else {
+                    return false;
+                }
+            }
+        }
+        if (rawSegment == null) {
```

Comment:
I don't understand this check, would `rawSegment` not always be null?

Reply:
It could be null if there are no raw segments. It could not be null if there is a single raw segment combined with only query segments.

Reply:
yeah but the assignment on line 209 only assigns it if it is `null` so what is the case where it is not `null`?

Reply:
its not null if it there is more than 1 raw segment